lib/pengine/utils.c

```diff
@@ -1263,9 +1264,28 @@ unpack_operation(pe_action_t * action, xmlNode * xml_obj, pe_resource_t * contai
         }
     }
 
-    value = g_hash_table_lookup(action->meta, XML_ATTR_TIMEOUT);
+    value = NULL;
+    if (pcmk__str_any_of(action->task, RSC_START, RSC_STATUS, NULL)
+        && action->rsc->parameters && action->rsc->xml
+        && safe_str_eq(crm_element_value(action->rsc->xml,
+                                         XML_AGENT_ATTR_CLASS),
+                       PCMK_RESOURCE_CLASS_STONITH)) {
+
+        const char *st_mon_timeout;
+        st_mon_timeout = g_hash_table_lookup(action->rsc->parameters,
+                                             "pcmk_monitor_timeout");
+        if (st_mon_timeout) {
+            value = st_mon_timeout;
+        }
+    }
+
+    if (!value) {
+        value = g_hash_table_lookup(action->meta, XML_ATTR_TIMEOUT);
```

Comment:
We can modify this a little bit if we want to include a debug message saying that we're overriding the meta timeout with `pcmk_monitor_timeout`.

Reply:
Definitely

Reply:
Ack. This is in place locally and will be in the next push to GitHub.